app/src/hooks/useShare.ts

```diff
@@ -0,0 +1,23 @@
+import { Share } from 'react-native';
+
+export const useShare = () => {
+  const share = async (code: string | undefined) => {
```

Comment:
Teraz jak exportujesz funkcję `share` to nie musisz zakładać, że `code` może być undefined - sprawdzanie tego warunku można zrobić w componencie. Dodatkowo sharować można każdy text a nie tylko kod zaproszenia więc może lepszą nazwą parametru byłoby `text`